runtime/autoload/health.vim

```diff
@@ -145,289 +139,56 @@ function! s:echo_notes(notes) abort
-          " that changes $PATH, like homebrew.
-          call add(notes, printf('Suggestion: There are multiple %s executables found.  '
-                \ . 'Set "g:%s" to avoid surprises.', python_bin_name, host_prog_var))
-        endif
-
-        if python_bin =~# '\<shims\>'
-          call add(notes, printf('Warning: "%s" appears to be a pyenv shim.  '
-                \ . 'This could mean that a) the "pyenv" executable is not in '
-                \ . '$PATH, b) your pyenv installation is broken.  '
-                \ . 'You should set "g:%s" to avoid surprises.',
-                \ python_bin, host_prog_var))
-        endif
-      endif
-    endif
-  endif
-
-  if !empty(python_bin)
-    if !empty(pyenv) && !exists('g:'.host_prog_var) && !empty(pyenv_root)
-          \ && resolve(python_bin) !~# '^'.pyenv_root.'/'
-      call add(notes, printf('Suggestion: Create a virtualenv specifically '
-            \ . 'for Neovim using pyenv and use "g:%s".  This will avoid '
-            \ . 'the need to install Neovim''s Python client in each '
-            \ . 'version/virtualenv.', host_prog_var))
-    endif
-
-    if !empty(venv) && exists('g:'.host_prog_var)
-      if !empty(pyenv_root)
-        let venv_root = pyenv_root
-      else
-        let venv_root = fnamemodify(venv, ':h')
-      endif
-
-      if resolve(python_bin) !~# '^'.venv_root.'/'
-        call add(notes, printf('Suggestion: Create a virtualenv specifically '
-              \ . 'for Neovim and use "g:%s".  This will avoid '
-              \ . 'the need to install Neovim''s Python client in each '
-              \ . 'virtualenv.', host_prog_var))
-      endif
-    endif
-  endif
-
-  if empty(python_bin) && !empty(python_bin_name)
-    " An error message should have already printed.
-    call add(notes, printf('Error: "%s" was not found.', python_bin_name))
-  elseif !empty(python_bin) && !s:check_bin(python_bin, notes)
-    let python_bin = ''
-  endif
-
-  " Check if $VIRTUAL_ENV is active
-  let virtualenv_inactive = 0
-
-  if exists('$VIRTUAL_ENV')
-    if !empty(pyenv)
-      let pyenv_prefix = resolve(s:trim(system(printf('"%s" prefix', pyenv))))
-      if $VIRTUAL_ENV != pyenv_prefix
-        let virtualenv_inactive = 1
-      endif
-    elseif !empty(python_bin_name) && exepath(python_bin_name) !~# '^'.$VIRTUAL_ENV.'/'
-      let virtualenv_inactive = 1
-    endif
-  endif
-
-  if virtualenv_inactive
-    call add(notes, 'Warning: $VIRTUAL_ENV exists but appears to be '
-          \ . 'inactive.  This could lead to unexpected results.  If you are '
-          \ . 'using Zsh, see: http://vi.stackexchange.com/a/7654/5229')
-  endif
-
-  " Diagnostic output
-  echo 'Checking: Python' a:version
-  echo '  Executable:' (empty(python_bin) ? 'Not found' : python_bin)
-  if len(python_multiple)
-    for path_bin in python_multiple
-      echo '     (other):' path_bin
+""
+" health#add_checker is a function to register a (or several) healthcheckers.
```

Comment:
Would it be better to replace "register a (or several)" with "register at least one"? Just a thought.